File: notebooks/survival_analysis/survival.R
library(tidyverse)
library(here)
library(yaml)

# here::i_am('survival.R')

PROJECT_PATH <- here()
config_path <- here(PROJECT_PATH, "config/config.yaml")
config <- read_yaml(config_path)

# Path

PROJECT_PATH <- config[['path_config']]['project_path']
INPUT_PATH <- here(PROJECT_PATH, 'data/processed/0_preprocess')
OUTPUT_PATH <- here(PROJECT_PATH, 'data/processed/notebooks')

d0 <- read_csv(here(INPUT_PATH, 'D0.csv'))


all <- d0 %>% colnames()


d0 <- d0 %>%
  group_by('PT_SBST_NO') %>%
  fill(all)


# 
d0 %>% 
  mutate(END=TIME) %>% head()

# Time Varying Cox regression analysis
library(survival)
tmerge(d0, tstop = TIME)


cgd0[,1:13] %>% head()

d0

coxph(Surv(tstart, tstop, DEAD_NFRM_DEAD))

File: notebooks/survival_analysis/survival.py
#%%
import pandas as pd
from pathlib import Path
import os, sys
from lifelines.utils import add_covariate_to_timeline

PROJECT_PATH = Path(__file__).parents[2].as_posix()
os.sys.path.append(PROJECT_PATH)
# %%
from src.MyModule.utils import *

config = load_config()
# %%
PROJECT_PATH = Path(config['path_config']['project_path'])
INPUT_PATH = PROJECT_PATH.joinpath('data/processed/0_preprocess')
OUTPUT_PATH = PROJECT_PATH.joinpath('data/processed/notebooks/')
# %%

d0 = pd.read_pickle(INPUT_PATH.joinpath('D0.pkl'))

idx = d0.PT_BSNF_BSPT_IDGN_AGE < 50
d1 = d0[idx].copy()
youngsters = d1.PT_SBST_NO.unique().tolist()
# %%
d1 = d0[d0.PT_SBST_NO.isin(youngsters)].copy()

# %% forward filling
patient_info = d1.PT_SBST_NO
d1 = d1.groupby(['PT_SBST_NO'], as_index=False).ffill()

#%%
d1['PT_SBST_NO'] = patient_info
# %%
d1
